DocumentService/Repositories/DocumentRepository.cs

```diff
@@ -0,0 +1,149 @@
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<bool> SetFileDeleted(Guid id, string userId)
+        {
+            try
+            {
+                DocumentInfo documentInfo = await this.GetDocument(id);
+
+                if (documentInfo != null)
+                {
+                    documentInfo.IsDeleted = true;
+                    documentInfo.DeletedById = userId;
+                    documentInfo.DateDeleted = DateTime.UtcNow;
+                    this.context.DocumentInfo.Update(documentInfo);
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<bool> Update(DocumentInfo documentInfo)
+        {
+            var updatedDocumentInfo = await this.GetDocument(documentInfo.DocumentId);
+            try
+            {
+                if (updatedDocumentInfo != null)
+                {
+                    updatedDocumentInfo.UserLastUpdatedById = documentInfo.UserLastUpdatedById;
+                    updatedDocumentInfo.FileName = documentInfo.FileName;
+                    updatedDocumentInfo.FileType = documentInfo.FileType;
+                    updatedDocumentInfo.Description = documentInfo.Description;
+                    updatedDocumentInfo.SubmissionMethod = documentInfo.SubmissionMethod;
+                    updatedDocumentInfo.Language = documentInfo.Language;
+                    updatedDocumentInfo.DocumentTypes = documentInfo.DocumentTypes;
+                    updatedDocumentInfo.DateLastUpdated = DateTime.UtcNow;
+
+                    this.context.DocumentInfo.Update(updatedDocumentInfo);
+                    return true;
+
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<DocumentInfo> GetDocumentsByIds(Guid[] ids)
+        {
+            return this.Filter(x => ids.Contains(x.DocumentId));
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<DocumentInfo> Filter(Expression<Func<DocumentInfo, bool>> predicate)
+        {
+            return context.DocumentInfo.Where(predicate).AsEnumerable<DocumentInfo>();
+        }
+
+        /// <summary>
+        /// Function to return a single document that is not set for deletion
+        /// </summary>
+        /// <param name="id">Id of document to find</param>
+        /// <returns>Document found</returns>
+        private async Task<DocumentInfo> GetDocument(Guid id)
+        {
+            var query = from q in context.DocumentInfo
```

Comment:
Can we change this to Linq as it more clean

Reply:
Done.